### ISSUE TITLE
Unable to fetch collections against local syncserver

### ISSUE THREAD

**Issue by callahad** [mozilla-services/syncserver#2]
## TL;DR

Using production Firefox Accounts against local syncserver fails: unable to fetch collections. Reports application crash, but appears to remain up and running.
## Context

The functional tests pass.
SyncServer is running on localhost:5000.
Firefox is set to use `http://localhost:5000/token/1.0/sync/1.5` as its `tokenServerURI`.
## Steps to Reproduce
1. Disconnect from Sync
2. In about:config, set `services.sync.tokenServerURI` to the above value. (Note: You have to do this every time. Disconnecting from Sync resets the `tokenServerURI`.)
3. Open the Browser Console, turn on Request/Response Body Logging, and sign into Sync
## What Happens:

```
GET https://accounts.firefox.com/config [HTTP/1.1 304 Not Modified 48ms]
POST https://api.accounts.firefox.com/v1/account/login [HTTP/1.1 200 OK 424ms]
GET https://api.accounts.firefox.com/v1/recovery_email/status [HTTP/1.1 200 OK 62ms]
GET https://api.accounts.firefox.com/v1/account/keys [HTTP/1.1 200 OK 83ms]
POST https://api.accounts.firefox.com/v1/certificate/sign [HTTP/1.1 200 OK 68ms]
GET http://localhost:5000/token/1.0/sync/1.5 [HTTP/1.0 200 OK 621ms]
GET http://localhost:5000/storage/1.5/1/info/collections [HTTP/1.0 503 Service Unavailable 21ms]
```

Followed by a JavaScript error:

```
A promise chain failed to handle a rejection.

Date: Thu Mar 20 2014 11:37:26 GMT-0500 (CDT)
Full Message: no user is logged in
Full Stack: JS frame :: resource://services-sync/browserid_identity.js :: this.BrowserIDManager.prototype.initializeWithCurrentIdentity/< :: line 182
JS frame :: resource://gre/modules/Promise.jsm :: Handler.prototype.process :: line 767
JS frame :: resource://gre/modules/Promise.jsm :: this.PromiseWalker.walkerLoop :: line 531
native frame :: <unknown filename> :: <TOP_LEVEL> :: line 0
```

The `GET http://localhost:5000/storage/1.5/1/info/collections` has a response body of:

```
"application error: crash id 6bea0e4a77b24a741d6990a7cb07cb24"
```

But as far as I can tell, the server is still up and running:

```
curl localhost:5000
it works!
```
